aplus/settings.py

```diff
@@ -174,6 +174,7 @@
     'social_django.middleware.SocialAuthExceptionMiddleware',
     'django.contrib.messages.middleware.MessageMiddleware',
     'django.middleware.clickjacking.XFrameOptionsMiddleware',
+    'aplus.middleware.LocaleMiddleware',
```

Comment:
I think this should be as early as possible (i.e. just after we have request.user). In addition, I think you should drop the LocaleMiddleware, so only our version is used.

Reply:
> In addition, I think you should drop the LocaleMiddleware, so only our version is used.

Is the `process_response` function of django's  `LocaleMiddleware` (https://docs.djangoproject.com/en/1.11/_modules/django/middleware/locale/) not in use or what should we do to it?

Reply:
I think the part

```python
patch_vary_headers(response, ('Accept-Language',)) # no path stuff
if 'Content-Language' not in response:
    response['Content-Language'] = language
```

would be relevant.

Reply:
Ok, I'll check what is needed from that and add it to our custom middleware.